src/planwise/client/scenarios/handlers.cljs

```diff
@@ -251,28 +251,34 @@
                       :new-provider :current-scenario
                       :get-suggestions :current-scenario
                       actual-state)]
-     {:db       (-> db (assoc :view-state next-state)
-                    (assoc-in [:current-scenario :suggested-locations] nil))
-      :dispatch (when getting-suggestions? [:scenarios.new-provider/abort-fetching-suggestions])})))
+     (merge
+      {:db       (-> db (assoc :view-state next-state)
+                     (assoc-in [:current-scenario :suggested-locations] nil))}
+      (when getting-suggestions?
+        {:dispatch  [:scenarios.new-provider/abort-fetching-suggestions]})))))
 
 (rf/reg-event-fx
  :scenarios.new-provider/fetch-suggested-locations
  in-scenarios
  (fn [{:keys [db]} [_]]
-   {:db  (-> db (assoc-in [:current-scenario :computing-best-locations :state] true)
-             (assoc :view-state :get-suggestions))
-    :api (assoc (api/suggested-providers (get-in db [:current-scenario :id]))
-                :on-success [:scenarios/suggested-providers]
-                :on-failure [:scenarios/no-suggested-providers])}))
+   (let [request-key (str (random-uuid))]
```

Comment:
We don't need a random id here. Since there can only be one in-flight request for suggestions, we can use some hard coded value for this key (eg. `:suggestions-request`).